Sources/ScipioKit/Producer/BinaryExtractor.swift

```diff
@@ -1,26 +1,29 @@
 import Foundation
-import PackageGraph
-import PackageModel
 import Basics
 
 struct BinaryExtractor {
-    var package: DescriptionPackage
+    var descriptionPackage: DescriptionPackage
     var outputDirectory: URL
     var fileSystem: any FileSystem
 
     @discardableResult
-    func extract(of binaryTarget: ScipioBinaryModule, overwrite: Bool) throws -> URL {
-        let sourcePath = binaryTarget.artifactPath
+    func extract(of binaryTarget: ResolvedModule, overwrite: Bool) throws -> URL {
+        guard case let .binary(binaryLocation) = binaryTarget.resolvedModuleType else {
+            preconditionFailure("\(#function) must be called with a binary target")
```

Comment:
The log message should contain the target's name and its actual module type to make the debugging easier.